tests/tasks/execute_with_reboot.yml

```diff
@@ -1,13 +1,33 @@
 - name: Execute the kdump role and reboot if necessary
   block:
-    - include_role:
+    - name: Find out reserved memory for the crash kernel
+      slurp:
+        src: /sys/kernel/kexec_crash_size
+      register: kexec_crash_size
+
+    - name: >-
+        If kdump has memory reserved for crash kernel, run the kdump role and reboot if necessary
+      include_role:
+        name: linux-system-roles.kdump
+      when: kexec_crash_size.content|b64decode|int > 0
+
+    - name: >-
+        If kdump has no memory reserved for crash kernel, set the
+        `tests:: reboot` tag, run the kdump role and reboot if necessary
+      include_role:
         name: linux-system-roles.kdump
+      when: kexec_crash_size.content|b64decode|int == 0
+      tags:
+        - tests::reboot
```

Comment:
tags are not set "if something", they are evaluated statically.

Reply:
not sure what you mean

but the `tests::reboot` tag is only used for testing currently, so that we can exclude entire tests - in general we can't rely on customer being able to set tags like this for specific runs of a role (e.g. when running via tower or some other execution environment) - that's why we prefer a solution where the user can control the behavior with a parameter set in an inventory, for example